### PR TITLE
[onert] helper funcs to check if shape has unspecified rank or dim

### PR DESCRIPTION
**Pull request by hyunsik-yoon**
This adds two helper functions to check if shape has unspecified rank or dim.

Signed-off-by: Hyun Sik Yoon <hyunsik.yoon.1024@gmail.com>

### CODE REVIEW

**Comment by hyunsik-yoon** [comment]
This will be used eventually to check unspecified rank or dim by `ANeuralNetworksExecution.cc`
For example,

https://github.com/Samsung/ONE/blob/master/runtime/onert/frontend/nnapi/wrapper/ANeuralNetworksExecution.cc#L101-L106

File: runtime/onert/core/src/ir/Shape.cc [hunk 1]
```diff
@@ -100,5 +100,13 @@ Shape permuteShape(const Shape &shape, Layout frontend_layout, Layout backend_la
   return backend_shape;
 }
 
+bool haveUnspecifiedDims(const ir::Shape &shape)
+{
```
Comment (ragmani) [inline runtime/onert/core/src/ir/Shape.cc:104]: If rank is 0, should this function return false?

Comment (hyunsik-yoon) [reply runtime/onert/core/src/ir/Shape.cc:104]: thanks! fixed!

Comment (ragmani) [reply runtime/onert/core/src/ir/Shape.cc:104]: I mean it would be `return true` if rank is 0

Comment (hyunsik-yoon) [reply runtime/onert/core/src/ir/Shape.cc:104]: It is the question of _"When rank is not specified, can we say dim is also not specified?"_ I think we'd better not to have overlapped area of both test. I will keep it and make better comment on `haveUnspecifiedDims`.

Comment (ragmani) [reply runtime/onert/core/src/ir/Shape.cc:104]: If so, the condition seems to be unnecessary. I'm sorry for confusion.

Comment (hyunsik-yoon) [reply runtime/onert/core/src/ir/Shape.cc:104]: haha~ you're correct. :-)

**Pull request merged**